pandas/core/arrays/interval.py

```diff
@@ -1862,11 +1862,17 @@ def isin(self, values) -> npt.NDArray[np.bool_]:
         return isin(self.astype(object), values.astype(object))
 
     @property
-    def _combined(self) -> IntervalSideT:
-        left = self.left._values.reshape(-1, 1)
-        right = self.right._values.reshape(-1, 1)
+    def _combined(self) -> IntervalSide:
+        # error: Item "ExtensionArray" of "ExtensionArray | ndarray[Any, Any]"
+        # has no attribute "reshape"  [union-attr]
+        left = self.left._values.reshape(-1, 1)  # type: ignore[union-attr]
+        right = self.right._values.reshape(-1, 1)  # type: ignore[union-attr]
```

Comment:
It might just always be an ndarray, at least I haven't found a way to instantiate IntervalArray with an ExtensionArray - will add and an assert and run the test locally.

Reply:
The tests reach it with a `DatetimeArray` which inherits reshape from `BaseMaskedArray`

Reply:
I think the only real alternative (without code changes) would be to investigate whether `Index._values` can have a narrowed type (not sure whether that is possible)

Reply:
Thanks for checking, certainly not something to be changing here.